### PR TITLE
Update button Catalog and Style documentation

### PR DESCRIPTION
**Pull request by MichelleGranat**
Added a small explanation of where the button style comes from.
Added an example of implementing button catalog for a custom theme.

I ran the example to make sure it works on 0.13

### CODE REVIEW

File: widget/src/button.rs [hunk 1]
```diff
@@ -505,6 +508,47 @@ impl Default for Style {
...
+///                 style.background = Some(Background::Color(Color::from_rgb(0.941, 0.502, 0.502)));
+///             },
+///         }
+/// 
+///         style
+///     }
+/// }
+/// ```
 pub trait Catalog {
     /// The item class of the [`Catalog`].
```
Comment (hecrj) [inline widget/src/button.rs:553]: May be interesting to mention here in the `Class` that `Button::style` can only be used if the associated type here implements: `From<StyleFn<'a, Self>>`.

File: widget/src/button.rs [hunk 2]
```diff
@@ -505,6 +508,47 @@ impl Default for Style {
...
 
 /// The theme catalog of a [`Button`].
+/// 
+/// All themes that can be used with [`Button`]
+/// must implement this trait.
+/// 
+/// # Example
+/// ```no_run
+/// #[derive(Debug, Default)]
+/// pub enum ButtonStyles {
```
Comment (hecrj) [inline widget/src/button.rs:518]: Maybe `ButtonClass`?

File: widget/src/button.rs (hunk 1 above)
Comment (MichelleGranat) [reply widget/src/button.rs:557]: Done

File: widget/src/button.rs (hunk 2 above)
Comment (MichelleGranat) [reply widget/src/button.rs]: Done

File: widget/src/button.rs [hunk 3]
```diff
@@ -505,6 +508,51 @@ impl Default for Style {
...
+///         }
+/// 
+///         style
+///     }
+/// }
+/// ```
+/// 
+/// Although, in order to use [`Button::style`]
+/// with `MyTheme`, [`Catalog::Class`] must implement
+/// `From<StyleFn<'_, MyTheme>`.
```
Comment (liigo) [inline widget/src/button.rs:555]: need one more `>`

Comment (MichelleGranat) [reply widget/src/button.rs]: Done

**Review by hecrj — APPROVED** [review]
Looks good! Thank you.

**Comment by MichelleGranat** [comment]
Fixed it and now it should pass the tests

**Pull request merged**